pkg/tsdb/azuremonitor/azmoncredentials/builder.go

```diff
@@ -134,7 +134,6 @@ func resolveLegacyCloudName(cloudName string) (string, error) {
 	case "":
 		return azsettings.AzurePublic, nil
 	default:
-		err := fmt.Errorf("the Azure cloud '%s' not supported by Azure Monitor datasource", cloudName)
```

Comment:
I think this error should remain. Data sources configured with custom clouds shouldn't hit this function as they should be created with the new credentials type. This error is valuable incase a legacy data source is misconfigured in some way.

Reply:
Ok, I have reverted this for now but Jeremy was hitting the legacy code path even for custom clouds, so we will have to see what we can do to make that work.

Reply:
With the update this should work as expected now